src/Be.Vlaanderen.Basisregisters.Projector/Internal/Runners/ConnectedProjectionsCatchUpRunner.cs

```diff
@@ -13,34 +14,83 @@ internal class ConnectedProjectionsCatchUpRunner
     {
         private readonly Dictionary<ConnectedProjectionName, CancellationTokenSource> _projectionCatchUps;
         private readonly IReadonlyStreamStore _streamStore;
-        private readonly IConnectedProjectionEventBus _eventBus;
+        private readonly IProjectionManager _projectionManager;
         private readonly ILogger<ConnectedProjectionsCatchUpRunner> _logger;
 
         public ConnectedProjectionsCatchUpRunner(
             IReadonlyStreamStore streamStore,
             ILoggerFactory loggerFactory,
-            IConnectedProjectionEventBus eventBus,
-            IConnectedProjectionEventHandler connectedProjectionEventHandler)
+            IProjectionManager projectionManager)
         {
             _projectionCatchUps = new Dictionary<ConnectedProjectionName, CancellationTokenSource>();
             _streamStore = streamStore ?? throw new ArgumentNullException(nameof(streamStore));
-            _eventBus = eventBus ?? throw new ArgumentNullException(nameof(eventBus));
+            _projectionManager = projectionManager ?? throw new ArgumentNullException(nameof(projectionManager));
             _logger = loggerFactory?.CreateLogger<ConnectedProjectionsCatchUpRunner>() ?? throw new ArgumentNullException(nameof(loggerFactory));
+        }
+
+        public bool IsCatchingUp(ConnectedProjectionName projectionName)
+        {
+            return projectionName != null &&
+                   _projectionCatchUps.ContainsKey(projectionName);
+        }
+
+        public void HandleCatchUpCommand<TCatchUpCommand>(TCatchUpCommand command)
+            where TCatchUpCommand : CatchUpCommand
+        {
+            _logger.LogTrace("CatchUp: Handling {Command}", command);
+            switch (command)
+            {
+                case StartCatchUp startCatchUp:
+                    Handle(startCatchUp);
+                    break;
+                case RemoveStoppedCatchUp removeStoppedCatchUp:
+                    Handle(removeStoppedCatchUp);
+                    break;
+                case StopCatchUp stopCatchUp:
+                    Handle(stopCatchUp);
+                    break;
+                case StopAllCatchUps _:
+                    StopAllCatchUps();
+                    break;
+                default:
+                    _logger.LogError("No handler defined for {Command}", command);
+                    break;
+            }
+        }
 
-            if (connectedProjectionEventHandler == null)
-                throw new ArgumentNullException(nameof(connectedProjectionEventHandler));
+        private void Handle(StopCatchUp stopCatchUp)
+        {
+            if (stopCatchUp.ProjectionName == null || false == IsCatchingUp(stopCatchUp.ProjectionName))
```

Comment:
flip `false ==`, or just `!IsCatchingUp`?

Reply:
flipped it, ! is nice and short but also easily missed when reading code